### ISSUE TITLE
Body not resetting to initial state when using updateBrowsers

### ISSUE THREAD

**Issue by Kemichal** [lihaoyi/workbench#16]
Hi, I'm new to scala.js and workbench so I might just have made a newbie mistake.

The workbench docs states this about updateBrowsers:
"Returning the state of document.body to the initial state before any javascript was run"
which I have been having troubles getting to work.

I have modified the master branch of workbench-example-app to replicate my issue and made a gif to show what is happening.

https://github.com/Kemichal/workbench-example-app

![workbench-issue](https://cloud.githubusercontent.com/assets/4427054/6009320/e0e7b6f8-ab28-11e4-9596-fd8e461c93a2.gif)

In short, when I change ScalaJSExample.scala workbench triggers and adds a child to body, but it does not return the state of body before doing so.

**Comment by LMnet** [comment]
I also have the same problem. I create a bunch of dom elements and after reload they append and append and append to the body without cleaning.

**Comment by Kemichal** [comment]
My issue boils down to shadowBody being lazily initiated after my main function have executed.

If you look at [WorkbenchClient.scala](https://github.com/lihaoyi/workbench/blob/ec99de45b82abdcc614e661afd1b9bb375ecd865/client/src/main/scala/workbench/WorkbenchClient.scala#L52-53 "WorkbenchClient.scala") you can see that the shadow copy is done *after* "Loading Workbench" is logged in the console. At this point my main function have already executed and is therefore part of the shadow body and will never disappear on update.

## Solution on application level

I solved this in my test application by changing

```html
<script>
    example.ScalaJSExample().main();
</script>
```
to
```html
<script>
    WorkbenchClient().shadowBody
    example.ScalaJSExample().main();
</script>
```
in index-dev.html.
Of coarse you can clear the body every time the main function is run, but with this approach the shadow body is bloated.

## Solutions on a library level
1. Change shadowBody to not be lazily initiated (works in my case, but will maybe not work for other cases).
2. Change documentation to indicate that you have to trigger the shadowBody initiation yourself.
3. Change the documentation about "Returning the state of document.body to the initial state before any javascript was run" to something more truthful.
4. Make Workbench itself execute the bootsnippet when it is setup and ready. The downside here is that there will be bigger differences between the dev html and the production html.

**Comment by ochrons** [comment]
Not relevant anymore.
